test/jit/test_cuda.py

```diff
@@ -0,0 +1,142 @@
+import os
+import sys
+
+import torch
+from torch.testing._internal.jit_utils import JitTestCase
+from torch.testing._internal.common_utils import skipIfRocm
+
+# Make the helper files in test/ importable
+pytorch_test_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(pytorch_test_dir)
+
+if __name__ == "__main__":
+    raise RuntimeError(
+        "This test file is not meant to be run directly, use:\n\n"
+        "\tpython test/test_jit.py TESTNAME\n\n"
+        "instead."
+    )
+
+class TestCUDA(JitTestCase):
+    """
+    A suite of tests for the CUDA API in TorchScript.
+    """
+
+    @skipIfRocm
+    def test_simple(self):
+        @torch.jit.script
+        def fn():
+            s = torch.classes.cuda.Stream(0, 0)
+            return False if s is None else True
```

Comment:
> I didn't look at the `torch/cuda/__init__.py` changes yet because I have to go work out but here is my initial feedback:
> 
> 1. We should aim to make sure the API for streams and events in TorchScript is as close as possible to eager mode (e.g. function names, argument names, argument order, defaults, etc.) even though we know we'll have a few small differences in the end.
> 2. I am not convinced the tests are thorough enough because there are no CUDA ops being invoked. I would suggest finding a CUDA op that has a high latency and using that to more thoroughly test event and stream synchronization, stream switching etc.
> 3. Some sort of comparison with code written in CUDA's eager mode API would be extremely valuable (although cumbersome at this point because we cannot script that eager mode code directly at the moment). It would be useful to make sure the numerical output of some computation expressed with streams and events and the event completion order is the same between eager mode and TorchScript.
> 4. We should add tests that check that TorchScript code that uses streams can be saved and loaded and executed in both Python and C++.

@SplitInfinity : I have done a few more changes.  Can you please review this ?
                        1. The JIT context manager support has been moved into torch/cuda/jit.py
                        2. Introduced a wrapper in jit.py to access torch.classes.cuda.Stream. Now the user can create the 
                            stream using torch.cuda.jit.Stream